controllers/workspace/status.go

```diff
@@ -31,6 +31,16 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 )
 
+const (
+	// devworkspacePhaseTerminating represents a DevWorkspace that has been deleted but is waiting on a finalizer.
+	// TODO: Should be moved to devfile/api side.
+	devworkspacePhaseTerminating dw.DevWorkspacePhase = "Terminating"
+
+	// devworkspacePhaseFailing represents a DevWorkspace that has encountered an unrecoverable error and will
```

Comment:
```suggestion
	// devworkspacePhaseFailing represents a DevWorkspace that has encountered an unrecoverable start-up error and will
```
Are we going to use it for anything else apart from startup failing?

Reply:
This is worth considering. I'd err on the side of generality, since it could potentially happen after a workspace is `Running` -- the logic applies based on whether or not `spec.started` is true. 

One scenario I could imagine us detecting in the future would be something like repeated OOMKilled in a workspace pod. Such a workspace would also require the `Failing` phase.